[Merged by Bors] - feat(measurable_space): more properties of measurable sets in a product

fpvandoorn opened the pull request:
Add multiple lemmas about `prod` to `set.basic`
Some cleanup in `set.basic`
Fix the name of the instance `measure_space ℝ`
Cleanup and a couple of additions to the `prod` section of `measurable_space`.
Rename: `prod_singleton_singleton` -> `singleton_prod_singleton`
Use `prod.swap` in the statement of `image_swap_prod`.

---
<!-- put comments you want to keep out of the PR commit here -->

fpvandoorn committed:
feat(measurable_space): more properties of measurable sets in a product

urkud commented on src/data/set/basic.lean:
```diff
@@ -1735,39 +1740,42 @@ ext $ by simp [set.prod]
 theorem insert_prod {a : α} {s : set α} {t : set β} :
   set.prod (insert a s) t = (prod.mk a '' t) ∪ set.prod s t :=
-ext begin simp [set.prod, image, iff_def, or_imp_distrib] {contextual := tt}; cc end
+begin ext ⟨x, y⟩, simp [set.prod, image, iff_def, or_imp_distrib, imp.swap] {contextual := tt} end
```
Do we have `prod_singleton`, `singleton_prod`, `prod_union`, and `union_prod`?

fpvandoorn replied:
I added them

urkud commented on src/data/set/basic.lean:
```diff
@@ -1735,39 +1740,42 @@ ext $ by simp [set.prod]
-  assume h, ⟨b, a, ⟨rfl, rfl⟩, h⟩⟩
+ext $ λ ⟨a, b⟩, by simp [mem_image_eq, set.prod, and_comm]; exact
+⟨ λ ⟨b', a', ⟨h_a, h_b⟩, h⟩, by subst a'; subst b'; assumption,
+  λ h, ⟨b, a, ⟨rfl, rfl⟩, h⟩⟩
```
If you move this below the next lemma, then the proof will become trivial. In particular, this is a way to get rid of a non-finalizing `simp`.

fpvandoorn replied:
I didn't even notice the nonterminal simp. Good catch!

urkud reviewed (commented):
LGTM

fpvandoorn committed:
more cleanup:

remove nonterminal simps
add more lemmas about prod
use projection notation for set.prod

fpvandoorn committed:
fix set/lattice

fpvandoorn committed:
lint fix

fpvandoorn committed:
remove 2 lemmas that were not very useful
